src/appleseed.shaders/include/appleseed/maya/as_maya_recurrence_helpers.h

```diff
@@ -0,0 +1,287 @@
+
+//
+// This source file is part of appleseed.
+// Visit https://appleseedhq.net/ for additional information and resources.
+//
+// This software is released under the MIT license.
+//
+// Copyright (c) 2018 Luis Barrancos, The appleseedhq Organization
+//
+// Permission is hereby granted, free of charge, to any person obtaining a copy
+// of this software and associated documentation files (the "Software"), to deal
+// in the Software without restriction, including without limitation the rights
+// to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+// copies of the Software, and to permit persons to whom the Software is
+// furnished to do so, subject to the following conditions:
+//
+// The above copyright notice and this permission notice shall be included in
+// all copies or substantial portions of the Software.
+//
+// THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+// IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+// FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+// AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+// LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+// OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+// THE SOFTWARE.
+//
+
+#ifndef AS_MAYA_FRACTAL_HELPERS_H
+#define AS_MAYA_FRACTAL_HELPERS_H
+
+#include "appleseed/math/as_math_helpers.h"
+#include "appleseed/maya/as_maya_recurrence_tables.h"
+
+//
+// Set of auxiliary functions for the Maya crater, stucco 3D texture node, see
+// also: ${MAYA_LOCATION}/docs/Nodes/crater.html
+//
+
+float recurrence(
+    float size,
+    point sample_center,
+    float shaker,
+    float melt)
+{
+    float tex_random_table[4096] = { RANDOM_TABLE_2 };
+    float tex_noise_offset[12] = { NOISE_TABLE_OFFSET };
+
+    float randomTable(int x, int y, int z)
+    {
+        return tex_random_table[
+             (x ^ y >> 4 ^ z >> 8) & 255 ^
```

Comment:
Please add back parentheses :) Just not the truly unnecessary ones like `(x)`.

Reply:
ok, i thought you meant simplifying to rely on the precedence rules only. Fixed.